tests/lib/devicetree/api/src/main.c

```diff
@@ -2225,6 +2225,31 @@ ZTEST(devicetree_api, test_parent)
 				  TEST_SPI_BUS_0), "");
 }
 
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT vnd_parent_bindings
+ZTEST(devicetree_api, test_parent_nodes_list)
+{
+	/* When traversing upwards, there are no fixed attributes and labels */
+	#define TEST_FUNC(parent) { /* No operation */ }
```

Comment:
I don't want to add too many properties for testing in the test dts, so I chose to do this. Make it only used to fill in the macro parameters, and then calculate whether the number of nodes it traverses matches.